server/src/main/java/org/opensearch/action/admin/cluster/stats/AggregatedNodeLevelStats.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.stats;
+
+import org.opensearch.action.admin.indices.stats.CommonStats;
+import org.opensearch.action.admin.indices.stats.ShardStats;
+import org.opensearch.action.support.nodes.BaseNodeResponse;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.index.cache.query.QueryCacheStats;
+import org.opensearch.index.engine.SegmentsStats;
+import org.opensearch.index.fielddata.FieldDataStats;
+import org.opensearch.index.shard.DocsStats;
+import org.opensearch.index.store.StoreStats;
+import org.opensearch.search.suggest.completion.CompletionStats;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Node level statistics used for ClusterStatsIndices for _cluster/stats call.
+ */
+public class AggregatedNodeLevelStats extends BaseNodeResponse {
```

Comment:
Default behaviour is as follows:

Coordinator node iterates over each `indices.stats.ShardStats` from each node response and creates a map of index_name to `ClusterIndicesStats.ShardStats`, i.e., for each index, the `indices.stats.ShardStats` returned from the nodes are iterated upon and added into the respective `ClusterIndicesStats.ShardStats` for that index. In the end, the `countsPerIndex` map contains indices and their accumulated `ClusterIndicesStats.ShardStats` made by adding up values from the indices.stats.ShardStats from each node

As this adds an additional overhead on the coordinating node of going through each `indices.stats.ShardStats` returned from each Node, we optimise it by computing `AggregatedNodeLevelStats` on each node and returning this itself.